storage/innobase/xtrabackup/src/ds_encrypt.c

```diff
@@ -552,6 +555,12 @@ encrypt_worker_thread_func(void *arg)
 		if (thd->cancelled)
 			break;
 
+		assert(gcry_md_get_algo_dlen(XB_CRYPT_HASH) ==
```

Comment:
I defined `XB_CRYPT_HASH_LEN` manually. It *must* match the length of the output given by chosen hash function (`XB_CRYPT_HASH`). I use it for buffer allocations and so on. This assert helps to ensure that if I change the hash function (`XB_CRYPT_HASH`), I also change `XB_CRYPT_HASH_LEN` and avoid buffer overflow errors. It helps to keep the code maintainable.